### PR TITLE
Kuzzle constructor update

### PR DESCRIPTION
**Pull request by scottinet**
Complies with the new Kuzzle object constructor format, taking a server address/IP instead of a full URL, and accepting a `port` option pointing to Kuzzle's REST port by default (7511)

### CODE REVIEW

File: src/Kuzzle.php [hunk 1]
```diff
@@ -101,6 +105,12 @@ public function __construct($url, array $options = [])
             $this->requestHandler = new CurlRequest();
         }
 
+        if (array_key_exists('port', $options)) {
+            $this->port = $options['port'];
+        }
+
+
+        $this->url = 'http://' . $host . ':' . $this->port;
```
Comment (ballinette) [inline src/Kuzzle.php:113]: should also set the protocol as an option, because we could (and should, in a production environment), put kuzzle behind a SSL proxy to reach it through https:// urls.

File: tests/KuzzleTest.php [hunk 2]
```diff
@@ -4,21 +4,22 @@
 
 class KuzzleTest extends \PHPUnit_Framework_TestCase
 {
+    const FAKE_KUZZLE_HOST = '127.0.0.1';
     const FAKE_KUZZLE_URL = 'http://127.0.0.1:7511';
-    
+
```
Comment (ballinette) [inline tests/KuzzleTest.php:9]: is FAKE_KUZZLE_URL still used ?

File: src/Kuzzle.php (hunk 1 above)
Comment (scottinet) [inline src/Kuzzle.php]: Good point. I'll also add SSL support to other SDKs.

File: tests/KuzzleTest.php (hunk 2 above)
Comment (scottinet) [inline tests/KuzzleTest.php:9]: Yes, in order to intercept the curl request and test if it's well formed.

File: src/Kuzzle.php (hunk 1 above)
Comment (scottinet) [inline src/Kuzzle.php:113]: Actually, adding SSL support involves more work than just using `https` in the URL: we also have to manage certificates, either to ignore them and/or (preferably "and") by allowing clients to use one.
I think that adding SSL support should be done in a separate PR.

Comment (ballinette) [inline src/Kuzzle.php:113]: ok

**Comment by ballinette** [comment]
+1

File: tests/KuzzleTest.php [hunk 3]
```diff
@@ -31,14 +32,14 @@ public function testSimpleConstructor()
             $this->assertEquals($routesNow, $routesDescription['read']['now']);
         }
         catch (Exception $e) {
-            $this->fail('KuzzleTest::testSimpleConstructor => Should not raise an exception (base configuration file should be valid)');
+            $this->fail('KuzzleTest::testSimpleConstructor => Should not raise an exception (base configuration file should be valid): ' . $e);
```
Comment (dbengsch) [inline tests/KuzzleTest.php]: $e->message would be better

**Comment by dbengsch** [comment]
+1
